From 495e8cffbf62f6809108442999084213b014e9e0 Mon Sep 17 00:00:00 2001
From: Alex Biddulph <abiddulph@4tel.com.au>
Date: Wed, 15 Nov 2023 08:59:44 +1100
Subject: [PATCH 1/2] Resolve compiler errors when compiling against Linux
 Kernel 6.6.

Specifically, the changes made are:
  - drm_gem_prime_fd_to_handle and drm_gem_prime_handle_to_fd are no longer exported as these are the defaults, so .prime_fd_to_handle and .prime_handle_to_fd no longer need to be set.
  - FBINFO_DEFAULT is no longer defined as it was defined as 0 and this is the default value (assuming the info struct is zero initialised)
---
 module/evdi_drm_drv.c | 4 ++++
 module/evdi_fb.c      | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/module/evdi_drm_drv.c b/module/evdi_drm_drv.c
index 09a9da0..b51bf2f 100644
--- a/module/evdi_drm_drv.c
+++ b/module/evdi_drm_drv.c
@@ -124,9 +124,13 @@ static struct drm_driver driver = {
 
 	.fops = &evdi_driver_fops,
 
+#if KERNEL_VERSION(6, 6, 0) > LINUX_VERSION_CODE
 	.prime_fd_to_handle = drm_gem_prime_fd_to_handle,
+#endif
 	.gem_prime_import = drm_gem_prime_import,
+#if KERNEL_VERSION(6, 6, 0) > LINUX_VERSION_CODE
 	.prime_handle_to_fd = drm_gem_prime_handle_to_fd,
+#endif
 #if KERNEL_VERSION(5, 11, 0) <= LINUX_VERSION_CODE || defined(EL8)
 #else
 	.preclose = evdi_driver_preclose,
diff --git a/module/evdi_fb.c b/module/evdi_fb.c
index 3789b33..3ae2ef6 100644
--- a/module/evdi_fb.c
+++ b/module/evdi_fb.c
@@ -418,7 +418,9 @@ static int evdifb_create(struct drm_fb_helper *helper,
 	info->fix.smem_start = (unsigned long)efbdev->efb.obj->vmapping;
 
 #if KERNEL_VERSION(4, 20, 0) <= LINUX_VERSION_CODE || defined(EL8)
+#ifdef FBINFO_DEFAULT
 	info->flags = FBINFO_DEFAULT;
+#endif
 #else
 	info->flags = FBINFO_DEFAULT | FBINFO_CAN_FORCE_OUTPUT;
 #endif
-- 
2.39.3

